### PR TITLE
[SYCL][HIP] Remove arg assert in LowerWGScope

### PR DESCRIPTION
**Pull request by AidanBeltonS**
This PR removes the assert for `parallel_for_work_items` with one argument. AMD correctly optimises away empty lambda arguments as a result only 1 argument is passed to the parallel_work_itme function. As a result this triggers the assert in `fixupPrivateMemoryPFWILambdaCaptures` as the number of arguments is equal to 1. 

This PR enables tests `SpecConstants/2020/kernel_lambda_with_kernel_handler_arg.cpp` and `SpecConstants/2020/non_native/aot_w_kernel_handler_wo_spec_consts.cpp` for the HIP backend.

### CODE REVIEW

**Review by v-klochkov — APPROVED** [review]

**Comment by bader** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/579

**Comment by bader** [comment]
@kbobrovs, several ESIMD tests fail with `error: GENX IR generation error: VISA builder API call failed: CisaBuilder->Compile("genxir", &ss, EmitVisa)`. Please, mark them as "expected to fail" while ESIMD team is working on the fix. This is needed to keep pre-commit results clean from false alarms.

@t4c1, please, apply the same for reduction tests impacted by https://github.com/intel/llvm/issues/5008.

**Comment by t4c1** [comment]
@bader I don't see any reduction tests failures with https://github.com/intel/llvm/pull/5025. I am not sure if marking some tests with xfail and reverting that after the PR is merged makes sense. The PR is complete, it is just waiting for review.

**Comment by bader** [comment]
Fixing failures is even better. Thanks!

**Comment by kbobrovs** [comment]
@bader - will do.

**Comment by kbobrovs** [comment]
> @bader - will do.

I can see these have already been marked. Working on a fix.

**Pull request merged**
